code/src/main/kotlin/com/expediagroup/sdk/core/logging/masking/LogMasker.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2024 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.sdk.core.logging.masking
+
+import com.ebay.ejmask.api.MaskingPattern
+
+/**
+ * Interface for pattern-based masking.
+ */
+internal interface PatternBasedMask : (String) -> String {
+    /**
+     * Adds masking patterns if they do not already exist.
+     *
+     * @param maskingPatterns Vararg of MaskingPattern to be added.
+     * @return True if patterns were added, false if they already existed.
+     */
+    fun addPatternIfNotExists(vararg maskingPatterns: MaskingPattern): Boolean
+
+    /**
+     * Retrieves the list of current masking patterns.
+     *
+     * @return A list of MaskingPattern.
+     */
+    fun patterns(): List<MaskingPattern>
+
+    /**
+     * Clears all existing masking patterns.
+     */
+    fun clear()
+}
+
+/**
+ * Object implementing the PatternBasedMask interface.
+ */
+internal val mask = object : PatternBasedMask {
```

Comment:
Why do we need this to be an object? Couldn't we build and pass the masking configs somewhere?